lib/adapters/QUnitAdapter.js

```diff
@@ -6,67 +6,37 @@ export default class QUnitAdapter extends EventEmitter {
     super()
     this.QUnit = QUnit
     QUnit.begin(this.onBegin.bind(this))
-    QUnit.moduleStart(this.onModuleStart.bind(this))
     QUnit.testStart(this.onTestStart.bind(this))
     QUnit.log(this.onLog.bind(this))
     QUnit.testDone(this.onTestDone.bind(this))
-    QUnit.moduleDone(this.onModuleDone.bind(this))
     QUnit.done(this.onDone.bind(this))
 
     this.testsDetails = {}
-    this.emitGlobal = false
+    this.type = {
```

Comment:
Let's try to get rid of the `type` argument entirely, and use separate `create` methods instead. `createTestStart` and `createTestEnd` should be obvious enough to implement. As for the other two, something like this, simplified using `map` instead of `let of / push`:

```js
createSuiteStart () {
  return new Suite(suite.name,
    suite.childSuites.map(this.createSuiteStart),
    suite.tests.map(this.createTestStart))
}

createSuiteEnd () {
  return new Suite(suite.name,
    suite.childSuites.map(this.createSuiteEnd),
    suite.tests.map(this.createTestEnd))
}
```
May need to explicitly bind both functions inside the constructor, to be able to pass them to `map` and still get the same `this` reference.